sharkfin/population.py

```diff
@@ -262,9 +253,10 @@ def init_simulation(self, T_sim=1000):
             agent.initialize_sim()
 
             if self.stored_class_stats is None:
-
                 ## build an IndShockConsumerType "double" of this agent, with the same parameters
-                ind_shock_double = cism.IndShockConsumerType(**agent.parameters)
+                parameters = agent.parameters.copy()
+                parameters["Rfree"] = parameters["Rfree"][0]
+                ind_shock_double = cism.IndShockConsumerType(**parameters)
```

Comment:
Not a HARK bug per se, but perhaps a design flaw...
Rfree is initially a float, but because of recent changes to HARK, it gets casted as a list, and a 1 element list in the case of infinite horizon. Then, when it is copied over, it is passed to this second agent as a list, which it does not like, so it has to be recast as a float again.

Reply:
Although maybe this case should be handled by HARK anyway.